File: tests/fixtures/edgecase_repo/dart/module1.dart
import 'dart:async';

import 'module0.dart';

abstract class Handler1 {
  bool handle(String payload);
}

class Base1 {
  final String name;
  Base1(this.name);
}

class Service1 extends Base1 implements Handler1 {
  Service1(String name) : super(name);

  @override
  bool handle(String payload) {
    this.validate(payload);
    return true;
  }

  void validate(String payload) {}
}

String helper1(String value) => value;

File: tests/fixtures/edgecase_repo/ruby/module1.rb
require_relative "./module0"

module Example
  class Base1
    def initialize(name)
      @name = name
    end
  end

  class Service1 < Base1
    def handle(payload)
      validate(payload)
      true
    end

    def validate(payload)
      !payload.nil?
    end
  end
end

def helper_1(value)
  value
end
